pandas/core/internals/managers.py

```diff
@@ -351,8 +351,25 @@ def where(self: T, other, cond, align: bool, errors: str) -> T:
             errors=errors,
         )
 
-    def setitem(self: T, indexer, value) -> T:
-        return self.apply("setitem", indexer=indexer, value=value)
+    def setitem(self: T, indexer, value, inplace: bool = False) -> T:
+        """
+        Set values with indexer.
+
+        For SingleBlockManager, this backs s[indexer] = value
+
+        Parameters
+        ----------
+        indexer, value
+        inplace : bool, default False
+            If True (for a Series), mutate the block/values in place, not
+            returning a new Manager/Block (and thus never changing the dtype).
+        """
+        if inplace:
+            # only passed for Series (single block)
+            assert self.ndim == 1
+            self._block.values[indexer] = value
```

Comment:
can these be unified by e.g. using self.arrays[0] here?

Reply:
Maybe yes, but in #41878 I would need to split the implementations then anyway (to add CoW handling in the AM version), so then it won't help with reducing the diff in that PR. 

Also, the version here uses `apply` and in Arraymanager it uses `apply_with_block`, so actually not that easy to combine.

Reply:
this will be sitting in master for a while, perhaps indefinitely.  let's do the de-duplication.

> it uses apply_with_block, so actually not that easy to combine

I think this has come up before: cant we alias BlockManager.apply_with_block=BlockManager.apply?

Reply:
I prefer to keep the apply_with_block in the ArrayManager, and not move them to the base class. It are cases that should still be updated. And I know that can take a while at the current tempo, but I don't see the value in temporarily moving them, while having them all in ArrayManager / single file is IMO easier to work with.

Reply:
I think the apply_with_block bit is no longer relevant.  Is there any reason not to share setitem_inplace between SingleArrayManager vs SingleBlockManager?

Reply:
Yep, by splitting it in two methods, the apply_with_block is indeed only relevant for the existing `setitem` which I don't touch anymore.

I moved `setitem_inplace` to the SingleDataManager base class.